src/accelerate/accelerator.py

```diff
@@ -480,6 +469,7 @@ def __init__(
                     "Tried to train with `fp8` and auto-detect backend, but no FP8-compatible backend was installed. "
                     "Valid backends are: `torchao`, `transformer-engine`, and `msamp`."
                 )
+            self.has_fp8_handler = True
```

Comment:
idk how it worked before but this was missing 😅

Reply:
if `has_fp8_handler` stays false, the property `fp8_backend` becomes wrong as it can only return MSAMP or None

Reply:
should I add some test for different expected values of fp8_backend ?